addnewcitydialog.cpp

```diff
@@ -5,78 +5,73 @@
 
 AddNewCityDialog::AddNewCityDialog(QWidget *parent) :
     QDialog(parent),
-    ui(new Ui::AddNewCityDialog),
-    m_messageBoxInfo{new QMessageBox(this)}
+    ui(new Ui::AddNewCityDialog)
 {
     ui->setupUi(this);
+
+    connect(ui->saveCancelButtonBox, &QDialogButtonBox::rejected, this, &AddNewCityDialog::onSaveCancelButtonBoxRejected);
+    connect(ui->saveCancelButtonBox, &QDialogButtonBox::accepted, this, &AddNewCityDialog::onSaveCancelButtonBoxAccepted);
 }
 
 AddNewCityDialog::~AddNewCityDialog()
 {
     delete ui;
 }
 
-void AddNewCityDialog::on_save_cancel_buttonBox_accepted()
+void AddNewCityDialog::onSaveCancelButtonBoxAccepted()
 {
-    if (ui->inputContinentComboBox->currentText() == "Select continent"
-        || ui->inputCountryLineEdit->text().isEmpty()
-        || ui->inputCityLineEdit->text().isEmpty()
-        || ui->inputPopulationLineEdit->text().isEmpty()
-        || ui->inputHighTempLineEdit->text().isEmpty()
-        || ui->inputLowTempLineEdit->text().isEmpty())
+    auto continent = ui->inputContinentComboBox->currentText();
+    auto country = ui->inputCountryLineEdit->text();
+    auto city = ui->inputCityLineEdit->text();
+    auto population = ui->inputPopulationLineEdit->text();
+    auto highestT = ui->inputHighTempLineEdit->text();
+    auto lowestT = ui->inputLowTempLineEdit->text();
+
+    if ( continent == "Select continent" || country.isEmpty() || city.isEmpty()
+        || population.isEmpty() || highestT.isEmpty() || lowestT.isEmpty())
     {
-        m_messageBoxInfo->setIcon(QMessageBox::Critical);
-        m_messageBoxInfo->setText(QString("Cannot add new item"));
-        m_messageBoxInfo->setInformativeText(QString("Empty fields were found, please fill all data"));
-        m_messageBoxInfo->setDefaultButton(QMessageBox::Ok);
-        m_messageBoxInfo->show();
+        QMessageBox *messageBoxInfo = new QMessageBox();
+        messageBoxInfo->setAttribute(Qt::WA_DeleteOnClose, true);
+        messageBoxInfo->setIcon(QMessageBox::Critical);
+        messageBoxInfo->setText(QString("Cannot add new item"));
+        messageBoxInfo->setInformativeText(QString("Empty fields were found, please fill all data"));
+        messageBoxInfo->setDefaultButton(QMessageBox::Ok);
+        messageBoxInfo->show();
         return;
     }
 
     bool wasConvertedPopulation(false), wasConvertedHighTemp(false), wasConvertedLowTemp(false);
     //if (ui->inputCountryLineEdit->text().toUInt(&wasConverted); !wasConverted)
-    ui->inputPopulationLineEdit->text().toUInt(&wasConvertedPopulation);
-    ui->inputHighTempLineEdit->text().toInt(&wasConvertedHighTemp);
-    ui->inputLowTempLineEdit->text().toInt(&wasConvertedLowTemp);
+    population.toUInt(&wasConvertedPopulation);
```

Comment:
Instead of these checks install QIntValidator and QDoubleValidator on top of QLineEdit

Reply:
done